tools/packaging/scripts/lib.sh

```diff
@@ -121,3 +122,14 @@ get_last_modification() {
 
 	echo "$(git log -1 --pretty=format:"%H" ${file})"
 }
+
+# $1 - The tag to be pushed to the registry
+# $2 - "yes" to use sudo, "no" otherwise
+push_to_registry() {
+	local tag="${1}"
+	local use_sudo="${2:-"yes"}"
+
+	if [ "${PUSH_TO_REGISTRY}" == "yes" ]; then
+		[ "${use_sudo}" == "yes" ] && sudo docker push ${tag} || docker push ${tag}
```

Comment:
This is a good example of `A && B || C` is not equal to `if A then B else C`. Because if B fails `sudo docker push ${tag}` then it will run C (`docker push ${tag}`) as well.

For example:
```
$ tag=foobar
$ use_sudo=yes
$ [ "${use_sudo}" == "yes" ] && sudo docker push ${tag} || docker push ${tag}
Using default tag: latest
The push refers to repository [docker.io/library/foobar]
An image does not exist locally with the tag: foobar
Using default tag: latest
The push refers to repository [docker.io/library/foobar]
An image does not exist locally with the tag: foobar

```

Reply:
Ack, expanded that locally!